code/datums/components/pitfall.dm

```diff
@@ -0,0 +1,199 @@
+/** There are many pitfalls in the game. Here is a list of them, for easier maintainence.
+ * Oshan/nadir trench:	1 direct drop, 1 area
+ * Polaris Pit:			2 landmarks
+ * sea elevator shaft:	1 landmark
+ * Icemoon:				2 landmarks
+ * Biodome:				3 landmarks
+ *
+ *
+ * These are ALL TURFS. They should STAY TURFS.
+ * similar but not quite the same as /datum/component/teleport_on_enter
+ *
+ * Here is a note preserved from the initial commit, referring to the icemoon abyss pitfall
+ * 	// this is the code for falling from abyss into ice caves
+ *	// could maybe use an animation, or better text. perhaps a slide whistle ogg?
+ **/
+
+ABSTRACT_TYPE(/datum/component/pitfall)
+/// A component for turfs which make movable atoms "fall down a pit"
+/datum/component/pitfall
+	/// a list of targets for the fall to pick from
+	var/list/TargetList = list()
+	/// the maximum amount of brute damage applied. This is used in random_brute_damage()
+	var/BruteDamageMax = 0
+	/// How long it takes for a thing to fall into the pit. 0 is instant, but usually you'd have a couple deciseconds where something can be flung across. Should use time defines.
+	var/FallTime = 0.3 SECONDS
+
+	// --------------- landmark targeting
+	/// The landmark that the fall sends you to. Should be a landmark define.
+	var/TargetLandmark = ""
+
+	// --------------- area targeting
+	/// The area path that the target falls into. For area targeting
+	var/TargetArea = null
+
+	// --------------- coordinate targeting
+	/// The z level that the target falls into if not via area or landmark.
+	var/TargetZ = 5
+	/// If truthy, try to find a spot around the target to land on in range(x).
+	var/LandingRange = 8
+
+
+/datum/component/pitfall/Initialize(BruteDamageMax = 50, FallTime = 0.3 SECONDS)
+	. = ..()
+	if (!istype(src.parent, /turf))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(src.parent, COMSIG_ATOM_ENTERED, PROC_REF(start_fall))
+	RegisterSignal(src.parent, COMSIG_ATTACKBY, PROC_REF(update_targets))
+	src.BruteDamageMax	= BruteDamageMax
+	src.FallTime		= FallTime
+	src.update_targets()
+
+
+/// returns the .parent but typecasted as a turf
+/datum/component/pitfall/proc/typecasted_parent()
+	RETURN_TYPE(/turf)
+	. = src.parent
+
+/// updates targets for area/coordinate targeting. is overridden added to in child types.
+/datum/component/pitfall/proc/update_targets()
+	return
+
+/// called when movable atom AM enters a pitfall turf.
+/datum/component/pitfall/proc/start_fall(var/signalsender, var/atom/movable/AM)
+	if (!istype(AM, /atom/movable) || istype(AM, /datum/projectile/))
```

Comment:
```suggestion
	if (!istype(AM, /atom/movable) || istype(AM, /obj/projectile))
```

Heh, funny how this was incorrect in the original code too. I guess that's where the bug report of neutrons falling down holes comes from.